src/storage/store/engine_tests.rs

```diff
@@ -1246,9 +1263,12 @@ mod tests {
         test_helper::assert_contains_all_messages(messages, &[cast3]);
 
         // We receive a merge event for the add and the intermediate remove, even though it would never get committed to the db
-        assert_merge_event(&event_rx.try_recv().unwrap(), &cast1, 0);
-        assert_merge_event(&event_rx.try_recv().unwrap(), &cast2, 1);
-        assert_merge_event(&event_rx.try_recv().unwrap(), &cast3, 2);
+        // First receive BLOCK_CONFIRMED event
+        let _block_confirmed_event = &event_rx.try_recv().unwrap();
```

Comment:
I think it makes sense to add a `recv_next_event` function which takes in an param to skip block confirmed events and events rx and skips the block confirmed events if indicated.